### PR TITLE
refactor(userspace/falco): re-design stats writer and make it thread-safe

### PR DESCRIPTION
**Pull request by jasondellaluce**
**What type of PR is this?**

/kind cleanup

/kind design

**Any specific area of the project related to this PR?**

/area engine

**What this PR does / why we need it**:

As for the requirements of https://github.com/falcosecurity/falco/issues/2074, this PR refactors the stats writer of Falco to make it thread-safe. The class has been re-designed as a whole with a multi-producer single-consumer model with a worker and a queue (similar to what we have in the output framework). Moreover, this approach should also perform a little better because during the event processing loop we now push on the queue instead of writing on the file buffer (it just happens once in a while when the periodic timeout ticks, so it's not a big deal). This design also make it easier to add new stats field if needed in the future.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
refactor(userspace/falco): re-design stats writer and make it thread-safe
```

### CODE REVIEW

**Review by FedeDP — COMMENTED** [review]
Very nice and clean PR! Thank you Jason!
I'll wait for approval until 0.32.1 is out :)

File: userspace/falco/stats_writer.cpp
Comment (leogr) [inline userspace/falco/stats_writer.cpp:2]:
```suggestion
Copyright (C) 2022 The Falco Authors.
```

File: userspace/falco/stats_writer.h
Comment (leogr) [inline userspace/falco/stats_writer.h:2]:
```suggestion
Copyright (C) 2022 The Falco Authors.
```

**Review by leogr — COMMENTED** [review]
SGMT, just the copyright year to be updated :)

**Review by FedeDP — APPROVED** [review]
/approve

**Review by leogr — APPROVED** [review]
Thank you

**Pull request merged**
